Cargo.toml

```diff
@@ -34,6 +34,7 @@ url = { version = "2.1", features = ["serde"] }
 uuid = { version = "0.8", features = ["serde", "v4"] }
 ws = { version = "0.9", features = ["ssl"] }
 zeroize = "1.1"
+ethers = "0.6.2"
```

Comment:
Not a huge fan of pulling in both `web3` and friends (`ethereum_types`, etc.) and `ethers`.

I'm fine with switching 100% to `ethers` though.

Reply:
We can use `ethers-core`.

But we need `ethers` if we want to switch from `web3`. Also there's a type from `ethereum_types` called `H520` which makes code simpler, otherwise we can use `Bytes` and implement `From<Bytes> for Signature`.

Reply:
Sounds great. What I understood from this was:
- In general, `ethereum-types` -> `ethers-core`
- Keep `ethereum-types` around for `H520` (personally I'm fine with this - its a fairly light-weight crate anyway).
- `web3` -> `ethers` (we can rename the `transport` feature to be `provider` so it makes more sense in Ethers-rs lingo)

For now, I published a `0.0.3` version with the fix (also updated some dependency versions and made most more liberal). That being said, I'd be happy to accept a PR for this. I just don't have the time to work on it personally.